### PR TITLE
Improve the VM's events' docs

### LINKED ISSUE #599: Documentation about awaiting async events
It might be my lack of Javascript knowledge but it took me a while to figure this out. The idea is to use the `step` event of the VM and make execute some async calls which we want to await before we continue executing the next step in the VM. An use case for this is to run a "handmade" mainnet fork to test some code on: copy the chain code and once it tries to read storage for anything (e.g. balance/`SLOAD`/`EXTCODECOPY`) we first read this from the blockchain itself, dump it in our VM chain and then continue executing.

From the docs I assumed that I could create an `async` event where I await these web3 calls - but this does *NOT* yield the VM.

The solution is to:

```
      function sleep(ms) {
        return new Promise(resolve => setTimeout(resolve, ms));
      }

      chain.on('step', async function (data, nextEventFunction) {
        await sleep(1000)
        nextEventFunction()
      })
```

The `nextEventFunction` MUST get called if it is provided as an argument in the function (if the argument length > 1 this MUST get called). 

This also has the weird side effect that if, for some reason, you create a function (does not even have to be `async`) which you attach to an event which has more than 1 argument in it, the VM will stop immediately after firing the event if you do NOT call the second argument.

### PR DESCRIPTION
**Pull request by alcuadrado**
Closes #599

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -156,9 +156,26 @@ You can subscribe to the following events of the VM:
...
+The first one will be the event object, and the second one a function.
+The VM won't continue until you call this function.
+
+If an exception is passed to that function, it will bubble into the
+VM and interrupt it, possibly corrupting its state. It's strongly
+recommended not to do that.
+
+### Synchronous event handlers
+
+If you want to perform asynchronous operations, you don't need
```
Comment (s1na) [inline README.md]: Should be synchronous.

File: README.md [hunk 2]
```diff
@@ -156,9 +156,26 @@ You can subscribe to the following events of the VM:
...
 
-If an exception is thrown from within an event handler, it will bubble into the VM and interrupt it, possibly corrupting its state. It's strongly recommended not to throw from event handlers.
+You can perform asynchronous operations from within an event handler
+and prevent the VM to keep running until they finish.
+
+In order to do that, your event handler has to accept two arguments.
+The first one will be the event object, and the second one a function.
+The VM won't continue until you call this function.
+
+If an exception is passed to that function, it will bubble into the
```
Comment (s1na) [inline README.md]: Does this only apply to async functions/

Comment (alcuadrado) [reply README.md]: You are right! It's the same for both.

File: README.md (hunk 1 above)
Comment (alcuadrado) [reply README.md]: Good catch!

**Pull request merged**